packages/components/Callout/macos/CalloutWindow.swift

```diff
@@ -26,11 +19,13 @@ class CalloutWindow: NSWindow {
 		backgroundColor = .clear
 		isMovable = false
 
-		NotificationCenter.default.addObserver(self, selector: #selector(appDidChangeActive(notification:)), name: NSApplication.didResignActiveNotification, object: nil)
+		// Dismiss the Callout if the window is no longer active.
+		NotificationCenter.default.addObserver(self, selector: #selector(dismissCallout), name: NSApplication.didResignActiveNotification, object: nil)
 
+		// Dismiss the Callout if the user touched/clicked outside the bounds of the callout.
```

Comment:
Yep, turns out it works. Both submenu and main menu get dismissed fine. Saves a bug for me to fix later lol

Reply:
Talked offline, and I should file a bug on this. It's possible to for a user to click on a submenu item (that shouldn't cause a dismiss because it's a custom item), and yet this code would still cause the whole callout to be dismissed.